mirgecom/wall_model.py

```diff
@@ -86,40 +225,85 @@ def tortuosity(self, tau: DOFArray) -> DOFArray:
         """Tortuosity of the porous material."""
         raise NotImplementedError()
 
+    @abstractmethod
+    def decomposition_progress(self, mass: DOFArray) -> DOFArray:
+        r"""Evaluate the progress ratio $\tau$ of the phenolics decomposition."""
+        raise NotImplementedError()
+
+
+class PorousFlowEOS:
+    """Abstract interface to equation of porous flow class."""
+
+    @abstractmethod
+    def decomposition_progress(self, material_densities) -> DOFArray:
+        r"""Evaluate the progress ratio $\tau$ of the phenolics decomposition."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def solid_density(self, material_densities) -> DOFArray:
+        r"""Return the solid density $\epsilon_s \rho_s$."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_temperature(self, cv: ConservedVars, wv: PorousWallVars,
+                        tseed: DOFArray, niter=3) -> DOFArray:
+        r"""Evaluate the temperature based on solid+gas properties."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_pressure(self, cv: ConservedVars, wv: PorousWallVars,
+                     temperature: DOFArray) -> DOFArray:
+        """Return the pressure of the gas considering the void fraction."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def internal_energy(self, cv: ConservedVars, wv: PorousWallVars,
+                 temperature: DOFArray) -> DOFArray:
+        """Return the enthalpy of the gas+solid material."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def heat_capacity(self, cv: ConservedVars, wv: PorousWallVars,
+                 temperature: DOFArray) -> DOFArray:
+        """Return the heat capacity of the gas+solid material."""
+        raise NotImplementedError()
+
+
+@dataclass(frozen=True)
+class PorousFlowModel(PorousFlowEOS):
```

Comment:
The `Model`/`EOS` terminology here is confusing me a little (especially with one inheriting from the other). 

Questions:

1. Is the base class needed? i.e., will there be more than one concrete class along the lines of `PorousFlowModel`?
2. If yes, should `PorousFlowModel` have a more specific name? It sounds kind of generic at the moment.

Reply:
I used `EOS` as the abstract class and the `model` as the wrapper class that combines both solid and gas (thus, a different EOS for the interacting solid+gas). The feeding classes (eos, transport and wall_model) are specific to gas, transport model and wall-only properties, respectively.

To clarify the individual classes, I added more details on the docs. If you have suggestions for variables/classes name in order to clarify, please!

1. I don't know if there will be, but it may be possible.
2. I used this name to keep the `model` in the nomenclature as in `GasModel`.

Let me know if it is still confusing.

Reply:
I would maybe drop the base class for now (unless there's a current use for it that I'm not quite understanding). Calling it an `EOS` is kind of confusing in a mirgecom context since the EOS is usually a member object of the `Model`, not a base class of it (or both, in this case).